compiler/src/iree/compiler/Codegen/Common/CPU/CPULowerToUKernels.cpp

```diff
@@ -461,10 +462,10 @@ matchDAGForUKernel(RewriterBase &rewriter, tensor::UnPackOp op,
 }
 
 static uint32_t
-getFlagForUserAndOperandTypes(IREE::LinalgExt::EncodingUser user,
+getFlagForUserAndOperandTypes(IREE::LinalgExt::EncodingAttr encoding,
                               ArrayRef<Attribute> operandTypes) {
-  if (user != IREE::LinalgExt::EncodingUser::MATMUL ||
-      operandTypes.size() != 3) {
+  auto cDims = getEncodingContractionDims(encoding);
+  if (failed(cDims) || !cDims->batch.empty() || operandTypes.size() != 3) {
     return IREE_UK_FLAG_QUERY_TILE_SIZES_OPERATION_NONE;
   }
```

Comment:
I think we should add a comment because the `user` is removed. And it was not clear to me why we have the check. Now I think I figure it out, the reason is that we only have ukernels for mmt4d ops but not batch_mmt4d ops. Can you add a related comment to it since you're touching this code?